presidio-anonymizer/presidio_anonymizer/entities/engine_request.py

```diff
@@ -0,0 +1,98 @@
+"""
+Engine request entity.
+
+It get the data and validate it before the engine receives it.
+"""
+import logging
+
+from presidio_anonymizer.anonymizers.fpe import FPE
+from presidio_anonymizer.anonymizers.hash import Hash
+from presidio_anonymizer.anonymizers.mask import Mask
+from presidio_anonymizer.anonymizers.redact import Redact
+from presidio_anonymizer.anonymizers.replace import Replace
+from presidio_anonymizer.entities.analyzer_result import AnalyzerResult
+from presidio_anonymizer.entities.analyzer_results import AnalyzerResults
+from presidio_anonymizer.entities.invalid_exception import InvalidParamException
+
+
+class AnonymizerEngineRequest:
+    """The request the engine will get and act on."""
+
+    anonymizers = {"mask": Mask, "fpe": FPE, "replace": Replace, "hash": Hash,
+                   "redact": Redact}
+
+    logger = logging.getLogger("presidio-anonymizer")
+
+    def __init__(self,
+                 data: dict):
+        """Handle text replacement for PIIs with requested transformations.
+
+        :param data: a map which contains the transformations, analyzer_results and text
+        """
+        self._transformations = {}
+        self._analysis_results = AnalyzerResults()
+        self.__validate_and_insert_input(data)
```

Comment:
Why do you need the double underscore here? Do you plan to have another colliding method with this name?